components/ws-daemon/pkg/internal/session/workspace.go

```diff
@@ -204,7 +204,12 @@ func (s *Workspace) Dispose(ctx context.Context) (err error) {
 	// old workspace content we can garbage collect that content later.
 	err = os.Remove(s.persistentStateLocation())
 	if err != nil {
-		return xerrors.Errorf("cannot remove workspace persistent state location: %w", err)
+		if os.IsNotExist(err) {
+			log.WithError(err).Warn("workspace persistent state location not exist")
+			err = nil
```

Comment:
I'm not sure should we keep running the following code?
https://github.com/gitpod-io/gitpod/blob/5fb85e47e307e8b2acc1ea5ac990ccd07abaceeb/components/ws-daemon/pkg/internal/session/workspace.go#L210-L232
or directly return at here.